XX.md

```diff
@@ -0,0 +1,266 @@
+
+## Events and Kinds
+
+### Product Listing (Kind: 30402)
+Following NIP-99's schema for product representation:
+
+```jsonc
+{
+  "kind": 30402,
+  "created_at": <unix timestamp>,
+  "content": "<product description in markdown>",
+  "tags": [
+    ["d", "<product identifier>"],
+    ["title", "<product title>"],
+    ["price", "<amount>", "<currency>", "<optional frequency>"],
+    // Optional tags
+    ["shipping", "<shipping-option-id>"], // References to shipping options
+    ["summary", "<short description>"],
+    ["image", "<url>", "<dimensions>"],
+    ["location", "<location string>"],
+    ["g", "<geo hash>"],
+    ["t", "<category>"],
+    ["a", "<product-list-id>"] // Reference to product collection if applicable
+  ]
+}
+```
+
+### Product Draft (Kind: 30403)
+Draft version of a product listing, following the same schema as Kind 30402:
+
+```jsonc
+{
+  "kind": 30403,
+  "created_at": <unix timestamp>,
+  "content": "<draft product description in markdown>",
+  "tags": [
+    // Same tag structure as Kind 30402
+  ]
+}
+```
+
+### Product Collection (Kind: 30405)
+Using NIP-51 list format for grouping products:
+
+```jsonc
+{
+  "kind": 30405,
+  "created_at": <unix timestamp>,
+  "content": "<optional collection description>",
+  "tags": [
+    ["d", "<collection identifier>"],
+    ["name", "<collection name>"],
+    // Optional tags
+    ["image", "<collection image>"],
+    ["summary", "<collection description>"],
+    ["location", "<location string>"],
+    ["a", "30402:<pubkey>:<product-id>"], // Product references
+    ["shipping", "<shipping-option-id>"] // References to shipping options
+  ]
+}
+```
+
+### Shipping Option (Kind: 30406)
+Dedicated event for shipping details:
+
+```jsonc
+{
+  "kind": 30406,
+  "created_at": <unix timestamp>,
+  "content": "<optional shipping description>",
+  "tags": [
+    ["d", "<shipping identifier>"],
+    ["name", "<shipping method name>"],
+    ["price", "<base_cost>", "<currency>"],
+    // ["location", "<service area>"],
+    // ["duration", "<estimated delivery time>"]
+    // Maybe we can use `L`and `l` tags for ontology, and use proper ISO3166 for shipping zones, also constraints for sizes and weight
+  ]
+}
+```
+
+## Order Communication Flow
+
+- Order processing uses [NIP-17](17.md) encrypted direct messages. Message direction is determined by the `p` tag - when sent from buyer to merchant, `p` contains the merchant's pubkey, and when sent from merchant to buyer, `p` contains the buyer's pubkey. 
+- The payment request message can be initiated in two ways, depending on whether the merchant has a server handling payments
+
+### Message Types
+
+1. Order Creation (buyer → merchant) (subject "order-info")
+```jsonc
+{
+  "kind": 14,
+  "tags": [
+    ["p", "<merchant-pubkey>"],
+    ["subject", "order-info"],
```

Comment:
`subject` is displayed as a title of the message, like the subject of an email. If "order-info" is a code that might not be human friendly, it might be better to choose a better tag name.